kong/router.lua

```diff
@@ -391,13 +391,20 @@ local function marshall_route(r)
         has_header_plain = true
 
         local header_values_map = {}
+        local header_values_count = 0
+        local header_pattern
         for i, header_value in ipairs(header_values) do
           header_values_map[lower(header_value)] = true
+          header_values_count = header_values_count + 1
+        end
+        if header_values_count == 1 and header_values[1]:find("~\\*") == 1 then
```

Comment:
the `if s:find(pfx) == 1 then...` pattern is not the best way to check if a string starts with a prefix, especially for external input data like here.  better to use `if s:sub(1,#pfx) == pfx then...`.

Also, `"~*"` is an odd prefix to use, very easy to get wrong, as happens here.  what about `"~~"`?

```suggestion
        if header_values_count == 1 and header_values[1]:sub(1,2) == "~~" then
```

Reply:
tks @javierguerragiraldez for your suggestion. I did modify it accordingly
The reason behind `"~*"` is that Nginx location block use that annotation for regex matching